script.js

```diff
@@ -1,5 +1,41 @@
 import data from './data.json' assert {type: 'json'};
 
+const buttonEl = document.querySelector(".filterBtnIcon")
+const filterPanelEl = document.querySelector(".filtersPanel")
+
+document.querySelector(".filterBtn").addEventListener('mouseenter', () => {
+    if (filterPanelEl.style.visibility === "hidden") { 
+        buttonEl.style.backgroundColor = "hsla(180, 28%, 50%, 1)";
+        buttonEl.innerHTML = '<img src="images/icon-up-hover.svg" />';
+    } else {
+        buttonEl.style.backgroundColor = "hsla(180, 28%, 50%, 1)";
+        buttonEl.innerHTML = '<img src="images/icon-down-hover.svg" />';
+    }
+})
+
+document.querySelector(".filterBtn").addEventListener('mouseleave', () => {
+    if (filterPanelEl.style.visibility === "hidden") { 
+        buttonEl.style.backgroundColor = "hsla(180, 31%, 95%, 1)";
+        buttonEl.innerHTML = '<img src="images/icon-up-active.svg" />';
+    } else {
+        buttonEl.style.backgroundColor = "hsla(180, 31%, 95%, 1)";
+        buttonEl.innerHTML = '<img src="images/icon-down-active.svg" />';
+    }
+})
+
+function toggleFilterBtn() {
+    if (filterPanelEl.style.visibility === "hidden") {
```

Comment:
tee siin
` console.log("filter panel visibiliy", filterPanelEl.style.visibility);`

Põhjus, miks esimesel korral ei avane, on see, et esimesel korral ei ole  `filterPanelEl.style.visibility` väärtust, on see, et sa proovid saada inline css väärtust, mida sul veel ei ole. 

https://zellwk.com/blog/css-values-in-js/

Sellel hetkel on sul visibility css kirjutatud ainult styles.css faili, selleks, et seda faili javascriptis lugeda, pead kasutama `getComputedStyle` funkstiooni. Ehk saad esimese visibility kätte  `getComputedStyle.(filterPanelEl).visibility)`

Teine variant on vaikimisi visibility stiil kirjutada otse HTMLi
` <div class="filtersPanel" style="visibility: hidden">`

Kolmas variant on teha `if` sees lisa-check, et see käivitada ka siis kui visibility stiili ei ole
` if (filterPanelEl.style.visibility === "hidden" || !filterPanelEl.style.visibility) { ...`